rules/prefer-starts-ends-with.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
+const doesNotContain = (string, chars) => chars.every(char => !string.includes(char));
+
+const isSimpleString = string => doesNotContain(
+	string,
+	['^', '$', '+', '[', '{', '(', '\\', '.', '?', '*']
```

Comment:
Thanks for the kind feedback, @jfmengels!

Just to make sure we are on the same page, here is what my code does currently: 

1. `/^foo/,` is correctly reported as _NOT valid_
2. `/\^foo/,` is correctly reported as _valid_
3. `/^\^foo/,` is reported as _valid_ (which is not ideal and as I had pointed it out in the PR description is what I am hoping to get help for)

Can you please confirm what we are looking to improve here is the 3rd case?

btw, just want to explain my thought process and to get feedback. My idea was to catch the use cases of `^ + simple string` and `simple string + $` and to deem `^ + not-so-simple string` as a legit use of regex (as `string#startsWith` doesn't cover such cases). My current way of determining whether a string is simple is obviously very naive. However, to handle cases like the 3rd case above, we have to be more aggressively categorizing a string as 'simple'. While we want to identify as many cases of misusing regex that can be replaced with `.startsWith` or `.endsWith` (by correctly categorizing more strings as 'simple') as possible, we probably want to be careful because if we wrongly categorizing a string as 'simple' the consequence is marking a legit use of regex as error(false positive), which could be worse than not reporting error on an improper use of regex(false negative).